week3/intro_to_costing.ipynb

```diff
@@ -0,0 +1,779 @@
+    "m.fs.properties.set_default_scaling(\"flow_mass_phase_comp\", 1, index=(\"Liq\", \"H2O\"))\n",
+    "m.fs.properties.set_default_scaling(\"flow_mass_phase_comp\", 1e2, index=(\"Liq\", \"TDS\"))\n",
+    "\n",
+    "# Set unit model values\n",
+    "set_scaling_factor(m.fs.pump.control_volume.work, 1e-3)\n",
+    "set_scaling_factor(m.fs.RO.area, 1e-2)\n",
+    "\n",
+    "# Calculate and propagate scaling factors\n",
+    "calculate_scaling_factors(m)\n",
+    "\n",
+    "# Get WaterTAP solver\n",
+    "solver = get_solver()\n",
+    "\n",
+    "# solve feed\n",
+    "solver.solve(m.fs.feed)\n",
+    "\n",
+    "# Propagate state from feed to pump\n",
+    "propagate_state(m.fs.feed_to_pump)\n",
+    "# Initialize pump\n",
+    "m.fs.pump.initialize()\n",
+    "\n",
+    "# Propagate state from pump to RO\n",
+    "propagate_state(m.fs.pump_to_RO)\n",
+    "# Initialize RO\n",
+    "m.fs.RO.initialize()\n",
+    "\n",
+    "# Propagate state from RO to brine and product\n",
+    "propagate_state(m.fs.RO_to_brine)\n",
+    "propagate_state(m.fs.RO_to_product)\n",
+    "\n",
+    "# Initialize brine and product\n",
+    "m.fs.brine.initialize()\n",
+    "m.fs.product.initialize()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c5dfb880-32ec-4432-98d7-71c2fb96b254",
+   "metadata": {},
+   "source": [
+    "## Step 3: Add RO Costing"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9d028cab-2dd0-40aa-be50-e3b93355ad73",
+   "metadata": {},
+   "source": [
+    "### Step 3a: Create the Costing Model\n",
+    "\n",
+    "`WaterTAPCosting` contain extensions, methods, and variables and constraints common to all WaterTAP Costing Packages. In other words, this sets up the base cost model, which we will expand upon by adding RO-specific cost terms."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "e8de9ef1-b78f-46e3-bb4d-f192dc4b6389",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set up WaterTAP Costing\n",
+    "m.fs.costing = WaterTAPCosting()\n",
+    "\n",
+    "# Set units of base currency\n",
+    "m.fs.costing.base_currency = pyunits.USD_2020"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "eee32511-d267-4023-a1c3-dfb928cbfefd",
+   "metadata": {},
+   "source": [
+    "### Step 3b: Add RO Unit Model Costing Block\n",
+    "Next, add the `UnitModelCostingBlock` for RO, which will set up all the RO-specific variables and constraints related to costing the unit. There are two configuration options on `UnitModelCostingBlock` that must be set. The first is `flowsheet_costing_block`, which refers to the instance of WaterTAP costing. This is conventionally referred to as `m.fs.costing`, and most flowsheets will only have one instance of WaterTAP costing; however, complicated flowsheets may contain multiple. The second configuration option is `costing_method`, which allows the user to choose between different cost assumptions. As shown  below, RO has two costing methods:\n",
```

Comment:
Also need to add a description for the costing_method_argument, assuming we want to leave that part in. I think if we don't include it, explaining how to update the RO membrane cost in step 4 will be confusing since it won't be clear when they decided to use high-pressure RO